src/UnigramWordPredictor.java

```diff
@@ -50,8 +52,14 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    */
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
-
     // TODO: Convert the trainingWords into neighborMap here
+    int counter = 0;
+    for(String item: trainingWords) {
+      counter++;
+      List<String> tempList = neighborMap.get(item);
+      tempList.add(trainingWords.get(counter));
+      neighborMap.put(item, tempList);
+    }
```

Comment:
I like the logic here, but there are some initialization issues. `neighborMap` never gets initialized and neither does a list when a entry is first put into the map for a string. This is what causes the tests to fail with null pointer exceptions. Make sure to run all the tests in the future and resolve any issues they flag.